include/electionguard/chaum_pedersen.hpp

```diff
@@ -6,10 +6,80 @@
 #include "group.hpp"
 #include "precompute_buffers.hpp"
 
+#include <map>
 #include <memory>
 
 namespace electionguard
 {
+    /// <summary>
+    /// Validation result for a zero knowledge proof
+    /// </summary>
+    struct ValidationResult {
```

Comment:
range proof returns a result with a collection of messages. this is currently only propagated in the cpp interface and not the c interface.